frame/support/src/weights/weight_v2.rs

```diff
@@ -93,27 +88,86 @@ impl Weight {
 		}
 	}
 
-	/// Construct with reference time weight.
+	/// Construct [`Weight`] with reference time weight.
 	pub const fn from_ref_time(ref_time: RefTimeWeight) -> Self {
 		Self { ref_time }
 	}
 
-	pub fn checked_mul(self, rhs: u64) -> Option<Self> {
-		let ref_time = self.ref_time.checked_mul(rhs)?;
-		Some(Self { ref_time })
+	/// Saturating [`Weight`] addition. Computes `self + rhs`, saturating at the numeric bounds of
+	/// all fields instead of overflowing.
+	pub const fn saturating_add(self, rhs: Self) -> Self {
+		Self { ref_time: self.ref_time.saturating_add(rhs.ref_time) }
+	}
+
+	/// Saturating [`Weight`] subtraction. Computes `self - rhs`, saturating at the numeric bounds
+	/// of all fields instead of overflowing.
+	pub const fn saturating_sub(self, rhs: Self) -> Self {
+		Self { ref_time: self.ref_time.saturating_sub(rhs.ref_time) }
+	}
+
+	/// Saturating [`Weight`] scalar multiplication. Computes `self.field * scalar` for all fields,
+	/// saturating at the numeric bounds of all fields instead of overflowing.
+	pub const fn saturating_mul(self, scalar: u64) -> Self {
+		Self { ref_time: self.ref_time.saturating_mul(scalar) }
 	}
 
-	pub fn checked_div(self, rhs: u64) -> Option<Self> {
-		let ref_time = self.ref_time.checked_div(rhs)?;
-		Some(Self { ref_time })
+	/// Saturating [`Weight`] scalar division. Computes `self.field / scalar` for all fields,
+	/// saturating at the numeric bounds of all fields instead of overflowing.
+	pub const fn saturating_div(self, scalar: u64) -> Self {
+		Self { ref_time: self.ref_time.saturating_div(scalar) }
 	}
 
-	pub const fn scalar_saturating_mul(self, rhs: u64) -> Self {
-		Self { ref_time: self.ref_time.saturating_mul(rhs) }
+	/// Saturating [`Weight`] scalar exponentiation. Computes `self.field.pow(exp)` for all fields,
+	/// saturating at the numeric bounds of all fields instead of overflowing.
+	pub const fn saturating_pow(self, exp: u32) -> Self {
+		Self { ref_time: self.ref_time.saturating_pow(exp) }
 	}
 
-	pub const fn scalar_div(self, rhs: u64) -> Self {
-		Self { ref_time: self.ref_time / rhs }
+	/// Increment [`Weight`] by `amount` via saturating addition.
+	pub fn saturating_accrue(&mut self, amount: Self) {
+		let mut o = Self::zero();
+		sp_std::mem::swap(&mut o, self);
+		*self = o.saturating_add(amount);
```

Comment:
```suggestion
		self.ref_time = amount.ref_time.saturating_add(amount.ref_time)
```

What am I missing?

Reply:
nothing, i had this fix, but forgot to push